### ISSUE TITLE
ProtectionHttpRequestHeaders isn't applied for video player when playing EAS stream

### ISSUE THREAD

**Issue by sturlath** [Megabit/Blazorise#4551]
If I try to setup and play a stream that is DASH with AES token the token isn´t applied.

```
<Video Source="@Video.DashUrl"
 StreamingLibrary="StreamingLibrary.Dash"
 ProtectionHttpRequestHeaders="@Video.Token" />
```

I just get a 400 error and a url that looks like this https://myserver-euno.streaming.media.azure.net/27531r60-8b40-4fef-b158-e00bd3ba45fc/96d76c64-3d5d-5255-8972-83cc89ed110e.ism/manifest(format=mpd-time-csf,encryption=cbc)&aes=true

As you can see there is no token being applied at the end. The stream is just fine and if I add the link to https://ampdemo.azureedge.net/ and add the token it just plays fine.                  

**Update**: I had forgotten[ I had asked this question](https://github.com/Megabit/Blazorise/issues/3846#issuecomment-1141845453) before 🙄

**Comment by sturlath** [comment]
I found [this conversation](https://github.com/Megabit/Blazorise/issues/3378#issuecomment-1049722855) we had @stsrki and I have been trying to get it to work but I just can´t get the token to show up.

Any ideas?

**Comment by sturlath** [comment]
I have been trying to load the Blazorise project so I can debug this and maybe do a PR but what ever I do VS just hangs.. 
funnily enough on the loading indicator project 😂
![image](https://user-images.githubusercontent.com/2386572/218084854-6d8db4df-0c7d-499a-8f28-e19fb0fee93e.png)

Any tips for me @stsrki if I manage to load it?

**Comment by stsrki** [comment]
That can happen sometimes but usually restart helps. Are you on latest VS? We use VS preview for the development.

**Comment by sturlath** [comment]
Yes I only use preview. Updated and restarted but still no go.. will try some stuff.

But regarding the issue, its a bug right? It should provide the token? If so Ill try to fix it.

**Comment by stsrki** [comment]
I haven't tried it so far, but I would say yes.

**Comment by sturlath** [comment]
OK finally managed to load it.. for others later go to `%USERPROFILE%\AppData\Local\Microsoft\VisualStudio` and delete all the cache folders there. 

I´ll start looking into the issue btw.

**Comment by sturlath** [comment]
I figured out the issue (I think). You are applying the protection data if there is any `VideoProtectionType` set. 

![image](https://user-images.githubusercontent.com/2386572/218110758-d88dd479-d2df-4653-9a14-84d9252d9615.png)

but I'm not using any DRM this time, just AES encryption.

```cshtml
<Video Source="@Video.DashUrl"
 StreamingLibrary="StreamingLibrary.Dash"
 ProtectionHttpRequestHeaders="@Video.Token" />
```

And thats the reason `Protection `part is never set.

I tried these steps without the token being set anywhere.
- Add AES to `VideoProtectionType` like `VideoProtectionType.AES`
- Update `ToVideoProtectionType()`
- Update the `OnAfterRenderAsync `logic

What am I missing?

**Comment by sturlath** [comment]
OK looks like[ dash.js doesn´t support AES](https://github.com/sampotts/plyr/issues/2620)... 

@johndeu does that mean we can´t use plyr.js for dash.js videos? What would you recommend using?
